service/src/main/java/uk/gov/hmcts/reform/fpl/handlers/NotificationHandler.java

```diff
@@ -195,4 +197,21 @@ private void sendCMONotificationForLocalAuthority(final CaseDetails caseDetails,
         sendNotification(CMO_ORDER_ISSUED_CASE_LINK_NOTIFICATION_TEMPLATE, email, localAuthorityNotificationParameters,
             Long.toString(caseDetails.getId()));
     }
+
+    private void sendCMODocumentLinkNotificationForCafcass(final CaseDetails caseDetails,
+                                                           final String localAuthorityCode,
+                                                           final DocmosisDocument document) {
+        Map<String, Object> cafcassParameters;
+        try {
+            cafcassParameters = caseManagementOrderEmailContentProvider.buildCMOIssuedNotificationParametersForCafcass(
+                caseDetails, localAuthorityCode, document);
+
+            String cafcassEmail = cafcassLookupConfiguration.getCafcass(localAuthorityCode).getEmail();
+
+            sendNotification(CMO_ORDER_ISSUED_DOCUMENT_LINK_NOTIFICATION_TEMPLATE, cafcassEmail, cafcassParameters,
+                Long.toString(caseDetails.getId()));
+        } catch (NotificationClientException e) {
```

Comment:
Keep try / catch block as small as possible. No point wrapping tens or hundreds of lines of code with try / catch if only one line can throw specific exception. As a reader I had to spend ages jumping through endless call delegations to finally locate source.